src/object/obj_tx.c

```diff
@@ -2836,9 +2862,12 @@ dc_tx_check_existence_task(enum obj_rpc_opc opc, daos_handle_t oh,
 	int					 rc;
 	int					 i;
 
+	tx->tx_has_cond = 1;
```

Comment:
[Suggestion] Similar to dc_tx_per_akey_existence_task, do this in dc_tx_attach while holding tx_lock.